### ISSUE TITLE
Consider using JSON convertion for path and query parameters

### ISSUE THREAD

**Issue by paulomorgado** [NSwag/NSwag#675]
When an **enum** value is passed in the path or query, the `EnumMemberAttribute` is ignored because the value is serialized with `ToString()`.

```
public enum Operation
{
    [EnumMember("set")]
    Set,
    [EnumMember("update")]
    Update,
}

public enum Value
{
    [EnumMember("one")]
    First,
    [EnumMember("two")]
    Second,
}

public async Task DoAsync(Operation operation, Value value)
{
    // path = {operation}
    // value is a query parameter
}
```

Called with `DoAsync(Operation.Set, Value.First)` will result in the `Set?First` URL instead of `set?one`.

It would be great if the JSON serializer was used to serialize values to path and query.

**Comment by paulomorgado** [comment]
Should this have a setting for retro-compatibility?

**Comment by RSuter** [comment]
I think it is fine to just fix it. If this is how web api behaves, the generated spec is always incorrect and we will not break someones code... Adding settings for every change makes the project/codebase too complicated...

**Comment by paulomorgado** [comment]
I was asking as it will break clients generate on previous versions that used, for example, a custom enum name generator.

**Comment by RSuter** [comment]
You are right but without without minor breaking changes it is very hard to evolve the project. And i think in this case it is fine because custom name generators are not that common and introducing legacy settings makes the codebase and usage more complicated.. what do you think?

**Comment by paulomorgado** [comment]
Json.NET doesn't seem to expose the enum conversion based on the `EnumMemberAttribute`. It will always be quoted.
